avocado/utils/process.py

```diff
@@ -1303,3 +1300,24 @@ def get_command_output_matching(command, pattern):
     :rtype: list of str
     """
     return [line for line in run(command).stdout_text.splitlines() if pattern in line]
+
+
+def get_ping_process_pid(interface="", peer_ip="", ping_count=""):
+    """
+    Funtion to get the process ID of ping flood.
+
+    :returns : Process PID number that initated by ping flood.
+    :rtype : int
+    """
+    cmd = (
+        f"ps -ef | grep '[p]ing -I {interface} {peer_ip} -c {ping_count} -f' "
+        f"| awk '{{print $2}}' | head -1"
```

Comment:
@FarooqAbdulla01 please check this command to get the PID easily and incorporate the same.

```
[root@sys ~]# ps -C ping
    PID TTY          TIME CMD
 284764 pts/1    00:00:00 ping
[root@sys ~]# 

```

This prints only the process that are running with specific command

here is the syntax

`ps -C <command>`

Reply:
@Naresh-ibm agreed, but i was looking for a solution which specifically check the PID for "ping" for peer machine of given IP address.

"ps -C ping" seems very generic and we cannot differentiate if there are multiple ping commands running on system.

In a multiprocess /multi test runs env's this may call to confusion over the PID.